pbj-core/buildSrc/src/main/kotlin/com.hedera.pbj.root.gradle.kts

```diff
@@ -14,12 +14,42 @@
  * limitations under the License.
  */
 
+import io.github.gradlenexus.publishplugin.CloseNexusStagingRepository
+
 plugins {
     id("com.hedera.pbj.repositories")
     id("com.hedera.pbj.aggregate-reports")
     id("com.hedera.pbj.spotless-conventions")
     id("com.hedera.pbj.spotless-kotlin-conventions")
     id("com.autonomousapps.dependency-analysis")
+    id("io.github.gradle-nexus.publish-plugin")
 }
 
+group = "com.hedera.pbj"
+
 spotless { kotlinGradle { target("buildSrc/**/*.gradle.kts") } }
+
+nexusPublishing {
+    repositories {
+        sonatype {
+            username = System.getenv("OSSRH_USERNAME")
+            password = System.getenv("OSSRH_PASSWORD")
+        }
+    }
+}
+
+tasks.withType<CloseNexusStagingRepository> {
+    // The publishing of all components to Maven Central (in this case only 'pbj-runtime') is
+    // automatically done before close (which is done before release).
+    dependsOn(":pbj-runtime:publishToSonatype")
+}
+
+tasks.register("releaseMavenCentral") {
+    group = "release"
+    dependsOn(tasks.closeAndReleaseStagingRepository)
+}
+
+tasks.register("releaseMavenCentralSnapshot") {
+    group = "release"
+    dependsOn(":pbj-runtime:publishToSonatype")
```

Comment:
Should this also be `dependsOn(tasks.closeAndReleaseStagingRepository)`? Or does the process work differently for the _Snapshot_ repo?

Reply:
The nexus plugin does not use a staging repository for snapshot builds. Even if we did add a `dependsOn` they would be skipped anyway by the Nexus plugin.

Reply:
Thanks for the explanation. Then we should keep it like this.